frontend/src/metabase/entities/bookmarks.js

```diff
@@ -46,9 +46,14 @@ const Bookmarks = createEntity({
     getIcon,
   },
   reducer: (state = {}, { type, payload, error }) => {
-    if (type === Questions.actionTypes.UPDATE && payload?.object?.archived) {
-      const key = `card-${payload.object.id}`;
-      return dissoc(state, key);
+    if (type === Questions.actionTypes.UPDATE && payload?.object) {
+      const { archived, id, name } = payload.object;
+      const key = `card-${id}`;
+      if (archived) {
+        return dissoc(state, key);
+      } else {
+        return updateIn(state, [key], item => ({ ...item, name }));
```

Comment:
Would be cool to also update `is_dataset` flag to handle when people turn questions into models and vice-versa.

Reply:
Nice one @kulyk, thanks.
I'll do it in a follow up PR and see about writing a test for it as well.